ChessVariantsLogic/Rules/Event.cs

```diff
@@ -0,0 +1,90 @@
+﻿
+using ChessVariantsLogic.Rules.Moves;
+using ChessVariantsLogic.Rules.Moves.Actions;
+using ChessVariantsLogic.Rules.Predicates;
+using ChessVariantsLogic.Rules.Predicates.ChessPredicates;
+
+namespace ChessVariantsLogic.Rules;
+/// <summary>
+/// Represents an event that can happen on the board. The Event will evaluate a predicate over the last boardTransition performed and then perform the internal set of actions.
+/// </summary>
+public class Event
+{
+    private readonly IPredicate _predicate;
+    private readonly ISet<IAction> _actions;
+    private readonly RelativeTo _relativeTo;
+
+    public Event(IPredicate predicate, ISet<IAction> actions, RelativeTo relativeTo)
+    {
+        _predicate = predicate;
+        _actions = actions;
+        _relativeTo = relativeTo;
+    }
+
+    /// <summary>
+    /// Evaluates if the internal predicate holds for the given BoardTransition.
+    /// </summary>
+    /// <returns>True if the predicate holds, otherwise false.</returns>
+    public bool ShouldRun(BoardTransition lastTransition)
+    {
+        return _predicate.Evaluate(lastTransition);
+    }
+
+    /// <summary>
+    /// Runs the event on the given <paramref name="moveWorker"/>. This does not take into account whether the event actually should be run.
+    /// </summary>
+    /// <param name="moveWorker">The MoveWorker we want to run the event on.</param>
+    /// <returns>True if the predicate holds, otherwise false.</returns>
+
+    public GameEvent Run(MoveWorker moveWorker)
+    {
+        Move lastMove = moveWorker.getLastMove();
```

Comment:
Assigns nullable type to non-nullable type

Reply:
Fixed now.